File: js/task3/S3_PIPE_05.js
// S3_PIPE_05

const logs = [
    "INFO: user=100 login success",
    "ERROR: db connection failed",
    "INFO: user=101 logout"
];

const extractUserIds = lines => {
    return lines
        .filter(line => line.startsWith("INFO:"))
        .map(line => {
            const parts = line.split(" ");
            const userPart = parts.find(p => p.startsWith("user="));
            return userPart ? userPart.split("=")[1] : null;
        });
};

console.log(extractUserIds(logs));

File: python/task3c/S3_PIPE_05.py
# S3_PIPE_05

def parse_logs(lines):
    ids = []
    for line in lines:
        if "INFO:" not in line:
            continue
            
        parts = line.split()
        for p in parts:
            if p.startswith("user="):
                try:
                    uid = int(p.split("=")[1])
                    ids.append(uid)
                except:
                    pass
    return ids

logs = [
    "INFO: user=100 login",
    "ERROR: fail",
    "INFO: user=101 logout"
]
print(parse_logs(logs))